pymeasure/instruments/hp/hp3478A.py

```diff
@@ -0,0 +1,427 @@
+        """Resets the display of the instrument if 1 is written to it.""",
+        set_process = (lambda x: int(x)),
+        validator = strict_discrete_set,
+        values = [1],
+        )
+
+    #TODO: find a proper validator for the next two methods (without using regex, if possible)
+    disp_text = Instrument.setting(
+        "D2%s",
+        """Displays up to 12 upper-case ASCII characters on the display.""",
+        set_process = (lambda x: str.upper(x[0:12])),
+        )
+
+    disp_text_no_symbol = Instrument.setting(
+        "D3%s",
+        """Displays up to 12 upper-case ASCII characters on the display and disables all symbols on the display.""",
+        set_process = (lambda x: str.upper(x[0:12])),
+        )
+
+    id = Exception(ValueError('Not implemented in this pre-SCPI instrument'))
+
+    #TODO: find a good way to cache the latest measurment and use it to incease speed (if actually needed)
+    measure_ACI = Instrument.measurement(
+        MODES["ACI"],
+        """Return the measured value for AC current""",
+        get_process = (lambda x: float(x)),
+        )
+
+    measure_ACV = Instrument.measurement(
+        MODES["ACV"],
+        """Return the measured value for AC Voltage""",
+        get_process = (lambda x: float(x)),
+        )
+
+    measure_DCI = Instrument.measurement(
+        MODES["DCI"],
+        """Return the measured value for DC current""",
+        get_process = (lambda x: float(x)),
+        )
+
+    measure_DCV = Instrument.measurement(
+        MODES["DCV"],
+        """Return the measured value for DC Voltage""",
+        get_process = (lambda x: float(x)),
+        )
+
+    measure_R2W = Instrument.measurement(
+        MODES["R2W"],
+        """Return the measured value for 2-wire resistance""",
+        get_process = (lambda x: float(x)),
+        )
+
+    measure_R4W = Instrument.measurement(
+        MODES["R4W"],
+        """Return the measured value for 4-wire resistance""",
+        get_process = (lambda x: float(x)),
+        )
+
+    measure_Rext = Instrument.measurement(
+        MODES["Rext"],
+        """Return the measured value for extended resistance mode (>30M, 2-wire) resistance""",
+        get_process = (lambda x: float(x)),
+        )
+
+    mode = Instrument.control(
+        "B",
+        "%s",
+        """Return current selected measurment mode, this propery can be set""",
+        get_process = (lambda x: get_mode(x.encode(encoding="ASCII"))),
+        set_process = (lambda x: MODES[x]),
+        validator = strict_discrete_set,
+        values = MODES,
+        )
+
+    #TODO: fix validators
+    range = Instrument.control(
+        "B",
+        "%s",
+        """Returns the current measurment range, this property can be set""",
+        get_process = (lambda x: get_range(x.encode(encoding="ASCII"))),
+        set_process = (lambda x,y: RANGES[x][y]),
+        # validator = joined_validators(strict_discrete_set,strict_discrete_set),
+        # values = RANGES,
```

Comment:
When using the `joined_validators`, the the argument for `values` is also expected to be a list that is equally long as the list of validators you pass to `joined_validators` (i.e. 2 long in this case)

Reply:
For the time being I have removed the validators complete from the range property.
My problem with this property is that it is depending on the measurement mode, I have added a set-function as an intermediate, but I am not happy with that yet.